From 00936136b07041d24b7497c8ef3a0fdcdefc57a9 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Mon, 15 May 2017 19:55:43 +0300
Subject: [PATCH 0084/1239] arm64: dts: a70x0: add a7040-db device tree for
 setup E

Update setup E of a7040-db according to board topology:
SerDes:
	- lane0: PCIe0
	- lane1: SATA0
	- lane2: SGMII0 (1G)
	- lane3: SATA1
	- lane4: USB1
	- lane5: PCIe1
AP-MPPs:
	- SP0, I2C, UART
CP-MPPs:
	- RGMII1, SDIO, 4xUART

Change-Id: I9ffd40c4d60aae9dfcfb19841af7b50427ef94f7
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Signed-off-by: Ken Ma <make@marvell.com>
---
 arch/arm/dts/Makefile             |   1 +
 arch/arm/dts/armada-7040-db-E.dts | 204 ++++++++++++++++++++++++++++++
 2 files changed, 205 insertions(+)
 create mode 100644 arch/arm/dts/armada-7040-db-E.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 32e3149203..ee19d2394b 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -186,6 +186,7 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-7040-db-B.dtb			\
 	armada-7040-db-C.dtb                    \
 	armada-7040-db-D.dtb			\
+	armada-7040-db-E.dtb			\
 	armada-7040-db-nand.dtb			\
 	armada-8040-db.dtb			\
 	armada-8040-db-B.dtb			\
diff --git a/arch/arm/dts/armada-7040-db-E.dts b/arch/arm/dts/armada-7040-db-E.dts
new file mode 100644
index 0000000000..cf95e2634a
--- /dev/null
+++ b/arch/arm/dts/armada-7040-db-E.dts
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2018 Marvell International Ltd.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 Development board platform
+ * Boot device: SPI NOR, 0x32 (SW3)
+ */
+
+#include "armada-70x0-db.dtsi"
+
+/ {
+	model = "Marvell Armada 7040 Development Board DB-88FA7040 setup(E)";
+	compatible = "marvell,armada7040-db", "marvell,armada7040",
+		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		i2c0 = &cp0_i2c0;
+		spi0 = &cp0_spi1;
+	};
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+};
+
+&ap_spi0 {
+	status = "okay";
+};
+
+&ap_i2c0 {
+	status = "okay";
+};
+
+&ap_pinctl {
+	   /* MPP Bus:
+	    * SPI0 [0-3]
+	    * I2C0 [4-5]
+	    * UART0 [11,19]
+	    */
+		  /* 0 1 2 3 4 5 6 7 8 9 */
+	pin-func = < 3 3 3 3 3 3 0 0 0 0
+		     0 3 0 0 0 0 0 0 0 3 >;
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&cp0_pcie0 {
+	status = "okay";
+};
+
+&cp0_pcie2 {
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c0_pins>;
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_i2c1 {
+	status = "okay";
+};
+
+&cp0_pinctl {
+		/* MPP Bus:
+		 * RGMII0 [0-11]
+		 * SPI1   [13-16]
+		 * SATA1  [28]
+		 * UART0  [29-30]
+		 * SMI	  [32,34]
+		 * I2C1   [35-36]
+		 * I2C0   [37-38]
+		 * UART1  [47-49]
+		 * UART0  [50-51]
+		 * SD	  [56-61]
+		 * GPIO   [62]
+		 */
+		/*   0   1   2   3   4   5   6   7   8   9 */
+	pin-func = < 3   3   3   3   3   3   3   3   3   3
+		     3   3   0   3   3   3   3   0   0   0
+		     0   0   0   0   0   0   0   0   9   0xA
+		     0xA 0   7   0   7   2   2   2   2   0
+		     0   0   0   0   1   1   1   7   1   7
+		     6   6   1   1   0xA 1   0xE 0xE 0xE 0xE
+		     0xE 0xE 0>;
+};
+
+&cp0_spi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_spi0_pins>;
+	status = "okay";
+
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor", "spi-flash";
+		reg = <0x0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "U-Boot";
+				reg = <0x0 0x200000>;
+			};
+
+			partition@400000 {
+				label = "Filesystem";
+				reg = <0x200000 0xe00000>;
+			};
+		};
+	};
+};
+
+&cp0_sata0 {
+	status = "okay";
+};
+
+&cp0_usb3_0 {
+	status = "okay";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&cp0_comphy {
+	phy0 {
+		phy-type = <COMPHY_TYPE_PEX0>;
+		phy-speed = <COMPHY_SPEED_5G>;
+	};
+
+	phy1 {
+		phy-type = <COMPHY_TYPE_SATA0>;
+	};
+
+	phy2 {
+		phy-type = <COMPHY_TYPE_SGMII0>;
+	};
+
+	phy3 {
+		phy-type = <COMPHY_TYPE_SATA1>;
+	};
+
+	phy4 {
+		phy-type = <COMPHY_TYPE_USB3_HOST1>;
+	};
+
+	phy5 {
+		phy-type = <COMPHY_TYPE_PEX2>;
+		phy-speed = <COMPHY_SPEED_5G>;
+	};
+};
+
+&cp0_utmi0 {
+	status = "okay";
+};
+
+&cp0_utmi1 {
+	status = "okay";
+};
+
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
+&cp0_mdio {
+	status = "okay";
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "sgmii"; /* lane-2 */
+};
+
+&cp0_eth1 {
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "rgmii-id"; /* lane-0 */
+};
-- 
2.29.0

